package.json

```diff
@@ -284,6 +284,7 @@
     "fast-json-patch": "3.1.0",
     "fast-text-encoding": "^1.0.0",
     "file-saver": "2.0.2",
+    "focus-trap-react": "^8.8.2",
```

Comment:
We already use `react-aria` for all things focus trapping in Grafana.
React-aria provides you with a set of hooks that helps you do things like focus trapping, keyboard navigation, `esc` closing, etc.
As for focus trapping, you can use the [FocusLock]( https://react-spectrum.adobe.com/react-aria/FocusScope.html) provided by react-aria to achieve similar result.

You can take a look at the [TimeRangePicker](https://github.com/grafana/grafana/blob/1aac13e5d0e703c9bc403c157e8af8e3025cf7ec/packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker.tsx#L120) component for some inspiration.

Reply:
@thisisobate thanks so much didn't notice this! Looks cool.